src/Model/Task/Task.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Freshheads Harvest API Client library.
+ *
+ * (c) Freshheads B.V. <info@freshheads.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace FH\HarvestApiClient\Model\Task;
+
+final class Task
+{
+    private $id;
+    private $name;
+    private $billableByDefault;
+    private $defaultHourlyRate;
+    private $isDefault;
+    private $isActive;
+    private $createdAt;
+    private $updatedAt;
+
+    public function __construct(
+        int $id,
```

Comment:
Ik heb bewust overal de constructor weggelaten. Je kunt namelijk ook een task aanmaken zonder id (`null`), die wordt in harvest automatisch gevuld. Daarnaast zijn ook niet alle properties required in de harvest API en kunnen er in vervolg versies andere properties bijkomen. Zaken zoals createdAt en updatedAt worden ook door harvest ingevuld

Reply:
@jvandesande De reden dat ik dit gedaan heb, is, dat we in ieder geval iets van gesloten models hebben. Anders moet je overal `=== null` checks toevoegen als je de models gebruikt binnen je applicatie. Dat haalt het hele nut van de models omver, en dan kun je misschien beter gewoon arrays of utility wrappers oid. terug gaan geven. (ook met het zicht op eventuele toevoegingen of veranderingen vanuit de API). De huidige strategie voegt iig. weinig toe.

Reply:
Op arrays kun je niet type hinten. Ik ben het met je eens dat het niet perse heel erg veel toevoegt, maar het voegt in ieder geval wel (type) validatie toe op classes en properties. Als we constructors toevoegen, zouden we alleen de in harvest verplichte velden moeten checken.

Toevoegingen aan de API kunnen we in principe gewoon meenemen als optionele properties.

Reply:
Ik maak er wel een soort van active record setup van dan :|